src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/refreshanalyzer/RefreshSearchAnalyzerResponse.kt

```diff
@@ -0,0 +1,136 @@
+ */
+
+package com.amazon.opendistroforelasticsearch.indexmanagement.refreshanalyzer
+
+import org.apache.logging.log4j.LogManager
+import org.elasticsearch.action.support.DefaultShardOperationFailedException
+import org.elasticsearch.action.support.broadcast.BroadcastResponse
+import org.elasticsearch.common.io.stream.StreamInput
+import org.elasticsearch.common.io.stream.StreamOutput
+import org.elasticsearch.common.xcontent.ConstructingObjectParser
+import org.elasticsearch.common.xcontent.ToXContent.Params
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.rest.action.RestActions
+import java.io.IOException
+import java.util.function.Function
+
+class RefreshSearchAnalyzerResponse : BroadcastResponse {
+
+    private var results: MutableMap<String, List<String>> = HashMap()
+    private var shardFailures: MutableList<FailedShardDetails> = mutableListOf()
+    private var temp: List<DefaultShardOperationFailedException> = mutableListOf()
+
+    protected var logger = LogManager.getLogger(javaClass)
+
+    @Throws(IOException::class)
+    constructor(inp: StreamInput) : super(inp) {
+        val resultSize: Int = inp.readVInt()
+        for (i in 0..resultSize) {
+            results.put(inp.readString(), inp.readStringArray().toList())
+        }
+
+        val failureSize: Int = inp.readVInt()
+        for (i in 0..failureSize) {
+            shardFailures.add(FailedShardDetails(inp.readString(), inp.readInt(), inp.readString()))
+        }
+    }
+
+    constructor(
+        totalShards: Int,
+        successfulShards: Int,
+        failedShards: Int,
+        shardFailures: List<DefaultShardOperationFailedException>,
+        results: MutableMap<String, List<String>>
+    ) : super(
+            totalShards, successfulShards, failedShards, shardFailures
+    ) {
+        this.results = results
+        this.temp = shardFailures
+        for (failure in shardFailures) {
+            this.shardFailures.add(FailedShardDetails(failure.index()!!, failure.shardId(), failure.reason()))
+        }
+    }
+
+    @Throws(IOException::class)
+    override fun toXContent(builder: XContentBuilder, params: Params?): XContentBuilder? {
+        builder.startObject()
+        RestActions.buildBroadcastShardsHeader(builder, params, totalShards, successfulShards, -1, failedShards, null)
+
+        builder.startArray("_successful")
+        for (index in results.keys) {
+            builder.startObject()
+            val reloadedAnalyzers: List<String> = results.get(index)!!
+            builder.field("index", index)
+            builder.startArray("refreshed_analyzers")
+            for (analyzer in reloadedAnalyzers) {
+                builder.value(analyzer)
+            }
+            builder.endArray()
+            builder.endObject()
+        }
+        builder.endArray()
+
+        builder.startArray("_failed")
+        for (failure in shardFailures) {
+            builder.startObject()
```

Comment:
Can chain these `builder` calls to make it look a little cleaner:

```
builder.startObject()
    .field("index", failure.index)
    .field("shardId", failure.shardId)
    .field("failureReason", failure.failureReason)
    .endObject()
```

Reply:
sure, this code is removed however.